spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/DenseTensor.scala

```diff
@@ -35,14 +35,21 @@ private[tensor] class DenseTensor[@specialized(Float, Double) T: ClassTag](
   var nDimension: Int)(implicit ev: TensorNumeric[T])
   extends Tensor[T] {
 
+  override def isEmpty: Boolean = this.storage() == null || this.storage().length() == 0
+
+  override def isScalar: Boolean =
+    this.nDimension == 0 &&
```

Comment:
this.nDimension == 0 && this._storage.length() != 1
does such tensor valid?

Reply:
User passes in an empty storage?

Reply:
nDimension and storage length is sufficient to determine whether a tensor is scalar. I will remove the test on size.